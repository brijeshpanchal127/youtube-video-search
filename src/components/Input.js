import React, { useState } from 'react'
import { Form, Button } from "react-bootstrap"
import { useDispatch } from 'react-redux';
import { searchVideos } from '../actions/search';


export const Input = () => {
    const [text, setText] = useState("");
    const dispatch = useDispatch();
    const searchText = async () => {
        const action = await searchVideos(text);
        dispatch(action);
    }

    return (
        <div>
            <Form.Group controlId="formBasicEmail">
                <Form.Label style={{ fontWeight: "bold" }}>Search Video</Form.Label>
                <Form.Control type="video" placeholder="Enter video name" value={text} onChange={(e) => setText(e.target.value)} />
            </Form.Group>
            <Button variant="primary" onClick={() => searchText()}>
                Search
                </Button>
        </div>
    )
}
