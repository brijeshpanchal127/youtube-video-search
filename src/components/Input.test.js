import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Input} from './Input';

const mockStore = configureStore([]);

describe('<Input />', () => {

    let store;
    let component;

    it('should render with initial state from Redux store', () => {
        store = mockStore({
            searchResults: [],
            refinedResults: [],
            refineText: "",
            onlyHD: false
        });
        component = renderer.create(
            <Provider store={store}>
                <Input />
            </Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

});