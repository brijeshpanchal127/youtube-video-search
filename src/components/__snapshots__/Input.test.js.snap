// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render with initial state from Redux store 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicEmail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Search Video
    </label>
    <input
      className="form-control"
      id="formBasicEmail"
      onChange={[Function]}
      placeholder="Enter video name"
      type="video"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</div>
`;
